psci/PSCi.hs

```diff
@@ -380,17 +381,50 @@ printModuleSignatures :: P.ModuleName -> P.Environment -> PSCI ()
 printModuleSignatures moduleName env =
   PSCI $ let namesEnv = P.names env
              moduleNamesIdent = (filter ((== moduleName) . fst) . M.keys) namesEnv
+             typesEnv = P.types env
+             typeSynonymsEnv = P.typeSynonyms env
+             moduleTypes = (filter (\(P.Qualified maybeName _) -> maybeName == Just moduleName) . M.keys) typesEnv
+             dataConstructorsEnv = P.dataConstructors env
+             moduleDataConstructors = (filter (\(P.Qualified maybeName _) -> maybeName == Just moduleName) . M.keys) dataConstructorsEnv
              in case moduleNamesIdent of
                   [] -> outputStrLn $ "This module '"++ P.runModuleName moduleName ++"' does not export functions."
-                  _ -> ( outputStrLn
-                       . unlines
-                       . sort
-                       . map (showType . findType namesEnv)) moduleNamesIdent
-  where findType :: M.Map (P.ModuleName, P.Ident) (P.Type, P.NameKind, P.NameVisibility) -> (P.ModuleName, P.Ident) -> (P.Ident, Maybe (P.Type, P.NameKind, P.NameVisibility))
-        findType envNames m@(_, mIdent) = (mIdent, M.lookup m envNames)
-        showType :: (P.Ident, Maybe (P.Type, P.NameKind, P.NameVisibility)) -> String
-        showType (mIdent, Just (mType, _, _)) = show mIdent ++ " :: " ++ P.prettyPrintType mType
-        showType _ = error "The impossible happened in printModuleSignatures."
+                  _ -> do
+                    (outputStr   . unlines . sort . mapMaybe (showType typeSynonymsEnv . findType typesEnv)) moduleTypes
+                    (outputStr   . unlines . map showDataConstructor . sortBy compareDatatypes . map (findDataConstructor dataConstructorsEnv)) moduleDataConstructors
+                    (outputStrLn . unlines . sort . map (showNameType . findNameType namesEnv)) moduleNamesIdent
+
+  where findNameType :: M.Map (P.ModuleName, P.Ident) (P.Type, P.NameKind, P.NameVisibility) -> (P.ModuleName, P.Ident) -> (P.Ident, Maybe (P.Type, P.NameKind, P.NameVisibility))
```

Comment:
Could you please add a space between declarations here? I think it will make it slightly easier to read. Also, to be consistent with other code, please align the declarations with the `where`.